pkg/controller/install/webhook.go

```diff
@@ -162,6 +188,144 @@ func (i *StrategyDeploymentInstaller) createOrUpdateValidatingWebhook(ogNamespac
 			log.Warnf("could not update ValidatingWebhookConfiguration %s", webhook.GetName())
 			return err
 		}
+
+		// check if there are any CRD names present in conversionCRDs field and
+		// check if isSingletonOperator() is true since conversion webhooks are scoped to AllNamespace Operators only
+		if len(desc.ConversionCRDs) != 0 && isSingletonOperator(i) {
+			if err := createOrUpdateConversionCRDs(desc, webhook.Webhooks[0].ClientConfig, i); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// check if csv supports only AllNamespaces install mode
+func isSingletonOperator(i *StrategyDeploymentInstaller) bool {
+	// get csv
+	csv, err := i.strategyClient.GetOpLister().OperatorsV1alpha1().ClusterServiceVersionLister().ClusterServiceVersions(i.owner.GetNamespace()).Get(i.owner.GetName())
+	if err != nil {
+		log.Info("CSV %s not found, error: %s", csv.GetName(), err.Error())
+	}
+	// get namespaces
+	namespaces := csv.Spec.InstallModes
+	// check if AllNamespaces is supported and other install modes are not supported
+	supportsOnlyAllNamespaces := false
+	supportsAllNamespaces := false
+	supportsOtherNamespaces := false
+	for _, namespace := range namespaces {
+		if namespace.Type == "AllNamespaces" && namespace.Supported {
+			supportsAllNamespaces = true
+		}
+		if namespace.Type != "AllNamespaces" && namespace.Supported {
+			supportsOtherNamespaces = true
+		}
+	}
+	if supportsAllNamespaces && supportsOtherNamespaces == false {
+		supportsOnlyAllNamespaces = true
+	}
+
+	return supportsOnlyAllNamespaces
+}
+
+func createOrUpdateConversionCRDs(desc v1alpha1.WebhookDescription, clientConfig admissionregistrationv1.WebhookClientConfig, i *StrategyDeploymentInstaller) error {
+
+	// iterate over all the ConversionCRDs
+	for index, ConversionCRD := range desc.ConversionCRDs {
+
+		// check if CRD exists on cluster
+		crd, err := i.strategyClient.GetOpClient().ApiextensionsInterface().ApiextensionsV1().CustomResourceDefinitions().Get(context.TODO(), desc.ConversionCRDs[index], metav1.GetOptions{})
+
+		if err != nil {
+			log.Info("CRD not found %s, error: %s", desc.ConversionCRDs[index], err.Error())
+		}
+		log.Info("Found conversionCRDs %s", desc.ConversionCRDs[index])
+
+		// check if the CSV owns this CRD
+		foundCRD := false
+		csv, err := i.strategyClient.GetOpLister().OperatorsV1alpha1().ClusterServiceVersionLister().ClusterServiceVersions(i.owner.GetNamespace()).Get(i.owner.GetName())
+		if err != nil {
+			log.Info("CSV not found, error: %s", err.Error())
+		}
+		ownedCRDs := csv.Spec.CustomResourceDefinitions.Owned
+		for _, ownedCRD := range ownedCRDs {
+			if ownedCRD.Name == crd.GetName() {
+				log.Info("CSV %q owns CRD %q", csv.GetName(), crd.GetName())
+				foundCRD = true
+			}
+		}
+		if !foundCRD {
+			log.Info("CSV %q does not own CRD %q", csv.GetName(), crd.GetName())
+			return nil
+		}
```

Comment:
Rather the querying for the CSV multiple times, could we get the CSV upfront and save a list of all the CRDs that it owns?